framework/src/interfacekernels/InterfaceReaction.C

```diff
@@ -0,0 +1,89 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "InterfaceReaction.h"
+
+registerMooseObject("MooseApp", InterfaceReaction);
+
+template <>
+InputParameters
+validParams<InterfaceReaction>()
+{
+  InputParameters params = validParams<InterfaceKernel>();
+  params.addParam<MaterialPropertyName>("D", "D", "The diffusion coefficient.");
```

Comment:
You don't appear to use these diffusion coefficients and it seems that in your documentation you mention that this object should be used in conjunction with an interface kernel like `InterfaceDiffusion`, so you should remove all the diffusion stuff from your class.